clusterman/autoscaler/pool_manager.py

```diff
@@ -203,6 +203,35 @@ def get_node_metadatas(self, state_filter: Optional[Collection[str]] = None) ->
             for instance_metadata in group.get_instance_metadatas(state_filter)
         ]
 
+    def terminate_expired_orphan_instances(self, dry_run: bool = False) -> None:
+        if dry_run:
+            logger.warning('Running in "dry-run" mode; cluster state will not be modified')
+
+        group_id_to_instance_ids = self.get_expired_orphan_instances()
+        logger.info(f"Terminating expired orphan instances: \n{group_id_to_instance_ids}")
+
+        if dry_run:
+            return
+
+        for group_id in group_id_to_instance_ids:
+            self.resource_groups[group_id].terminate_instances_by_id(group_id_to_instance_ids[group_id])
+
+    def get_expired_orphan_instances(self, timeout: int = 1800) -> Mapping[str, List[str]]:
```

Comment:
nit:
```suggestion
    def get_expired_orphan_instances(self, timeout: int = 1800) -> Dict[str, List[str]]:
```

we generally prefer to type return values with more concrete types rather than with interfaces